serf/serf.go

```diff
@@ -1242,14 +1242,15 @@ func (s *Serf) handleQuery(query *messageQuery) bool {
 
 	if s.config.EventCh != nil {
 		s.config.EventCh <- &Query{
-			LTime:    query.LTime,
-			Name:     query.Name,
-			Payload:  query.Payload,
-			serf:     s,
-			id:       query.ID,
-			addr:     query.Addr,
-			port:     query.Port,
-			deadline: time.Now().Add(query.Timeout),
+			LTime:       query.LTime,
+			Name:        query.Name,
+			Payload:     query.Payload,
+			serf:        s,
+			id:          query.ID,
+			addr:        query.Addr,
+			port:        query.Port,
+			deadline:    time.Now().Add(query.Timeout),
+			relayFactor: s.config.QueryResponseRelayLimit,
```

Comment:
We should pull this out of config and make it come from the `QueryParam` struct - that way the query-er gets to pick what happens. We'd also need to add that to the `serf query` CLI and API client. This'll give users full control of this feature. For Consul we can do the same when we integrate this. For keyring ops we can maybe just set it based on the size of the cluster.